include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -92,16 +92,17 @@ typename MatType::elem_type NSGA2::Optimize(
   numVariables = iterate.n_rows;
 
   // Cache calculated objectives.
-  std::vector<arma::Col<ElemType> > calculatedObjectives;
+  std::vector<arma::Col<ElemType> > calculatedObjectives; //TODO: Size is fixed, perhaps use matrix?
```

Comment:
The size of the objectives depends on the size of the population, which can increase in case of a crossover, so I don't think you can get away with a const matrix.